lib/utils/eol-conversion.ts

```diff
@@ -136,3 +117,123 @@ export async function readFileWithEolConversion(
 		return fileBuffer;
 	}
 }
+
+/**
+ * Attempt to detect the encoding of a data buffer.
+ * Code copied and modified from the npm package 'isbinaryfile' (MIT licence)
+ *   https://github.com/gjtorikian/isBinaryFile/blob/master/src/index.ts
```

Comment:
To emphasize it, this function was copied from the `isbinaryfile` npm module (MIT licence). The original code is at https://github.com/gjtorikian/isBinaryFile/blob/4.0.4/src/index.ts#L62

The modifications I made were:
* The original function would return true/false (true=binary), and the new function returns a string naming the detected encoding (with `'binary'` representing a generic binary encoding, and 'utf-8' encompassing 'ascii').
* The original function would probabilistically conclude that a file is binary if more than 10% of characters were in the ASCII ranges 0 to 7 and 14 to 32. Because of our purpose of "editing" text files to replace CRLF with LF, I changed the detection to conservatively conclude that a file is binary if even a single character is found in those ranges, **except for** character ESC (escape, number 27 = 0x1B) which is used in ANSI escape sequences for text color.
* I've added some test code, but I didn't worry too much about the UTF-8 algorithm details because I didn't change it, and the original code had already been tested by more than 2 million downloads a week...

Reply:
Btw, my first attempt was to add `isbinaryfile` to `package.json` and then patch the function with `patch-package`, but the patch file turned out to be larger than the original function and harder to read. So I thought I'd just copy-and-paste the function, and add some notes about it.

Reply:
In the end I didn't use `patch-package` -- actually this PR deletes a patch file that was being used to patch `mmmagic`. As for forking the module, I like better your other suggestion of creating a new small npm package dedicated to EOL conversion -- also because the "patch" to `isBinaryFile` isn't a bug fix, it actually slightly re-purposes the function/package to detecting encodings that are suitable for EOL conversion.  Maybe for this PR we can live with the function copy-and-paste, as we need to release a fix for Windows CLI users out there.